### PR TITLE
updating StatefulSet and Deployment API Version

### PR DESCRIPTION
**Pull request by jainishshah17**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/artifactory/templates/nginx-deployment.yaml [hunk 1]
```diff
@@ -66,9 +68,7 @@ spec:
               - '/bin/sh'
               - '-c'
               - >
-                {{- if .Values.nginx.customArtifactoryConfigMap }}
-                cp -Lrf /tmp/conf.d/artifactory*.conf /etc/nginx/conf.d/artifactory.conf;
```
Comment (danielezer) [inline stable/artifactory/templates/nginx-deployment.yaml]: How would this get copied not to /etc/nginx?

Comment (jainishshah17) [reply stable/artifactory/templates/nginx-deployment.yaml]: Directly mounting it there

File: stable/artifactory-ha/templates/nginx-deployment.yaml [hunk 2]
```diff
@@ -125,7 +125,8 @@ spec:
         {{- end }}
         {{- if .Values.nginx.customArtifactoryConfigMap }}
         - name: artifactory-nginx-config
-          mountPath: "/tmp/conf.d"
+          mountPath: "/tmp/conf.d/artifactory-ha.conf"
```
Comment (danielezer) [inline stable/artifactory-ha/templates/nginx-deployment.yaml]: It’s still mounted to /tmp

File: stable/artifactory/templates/nginx-deployment.yaml (hunk 1 above)
Comment (danielezer) [reply stable/artifactory/templates/nginx-deployment.yaml]: It’s still mounted to /tmp

File: stable/artifactory/templates/nginx-deployment.yaml [hunk 3]
```diff
@@ -125,7 +125,8 @@ spec:
         {{- end }}
         {{- if .Values.nginx.customArtifactoryConfigMap }}
         - name: artifactory-nginx-config
-          mountPath: "/tmp/conf.d"
+          mountPath: "/tmp/conf.d/artifactory.conf"
+          subPath: "artifactory.conf"
```
Comment (jainishshah17) [inline stable/artifactory/templates/nginx-deployment.yaml]: @danielezer here

**Comment by danielezer** [comment]
@jainishshah17 it’s pointing to /tmp, not to /etc/nginx. Or am I missing something?

File: stable/artifactory/templates/nginx-deployment.yaml (hunk 3 above)
Comment (jainishshah17) [reply stable/artifactory/templates/nginx-deployment.yaml]: @danielezer My bad. fixing it

File: stable/artifactory-ha/templates/nginx-deployment.yaml [hunk 4]
```diff
@@ -125,7 +125,8 @@ spec:
         {{- end }}
         {{- if .Values.nginx.customArtifactoryConfigMap }}
         - name: artifactory-nginx-config
-          mountPath: "/tmp/conf.d"
+          mountPath: "/etc/nginx/conf.d/artifactory-ha.conf"
+          subPath: "artifactory-ha.conf"
```
Comment (danielezer) [inline stable/artifactory-ha/templates/nginx-deployment.yaml]: Shouldn’t this be configurable? Or at least documented? I mean, the fact that you have to have a key called artifactory-ha.conf here and artifactory.conf in the artifactory chart

Comment (jainishshah17) [reply stable/artifactory-ha/templates/nginx-deployment.yaml]: @danielezer  https://github.com/jfrog/charts/blob/master/stable/artifactory-ha/ReverseProxyConfiguration.md#steps-to-use-static-configuration-for-reverse-proxy-in-nginx

Comment (danielezer) [reply stable/artifactory-ha/templates/nginx-deployment.yaml]: I see that but I think it’s worth mentioning that the data key has to be called by that specific name. I’m not I would understand that based in the current documentation. But that’s just me 😅

Comment (jainishshah17) [reply stable/artifactory-ha/templates/nginx-deployment.yaml]: @danielezer I do agree. Do you want me to make it configurable or just add note in the readme?

Comment (danielezer) [reply stable/artifactory-ha/templates/nginx-deployment.yaml]: Mmm whatever you think 🙂
I think both are good

**Comment by jainishshah17** [comment]
> I like this change. Good clean up.
> Does it require to update minimal K8s version supported?

It should work with all version.

File: stable/artifactory-ha/templates/nginx-deployment.yaml (hunk 4 above)
Comment (danielezer) [reply stable/artifactory-ha/templates/nginx-deployment.yaml]: @jainishshah17 what about this change? what do you think?

Comment (jainishshah17) [reply stable/artifactory-ha/templates/nginx-deployment.yaml]: @danielezer Done. PTAL.

**Pull request merged**